### PR TITLE
Api 13164/loast accept config file

### PR DESCRIPTION
**Pull request by peyton-lucas**
This branch allows LOAST to accept a config file via the PositiveAll command. PositiveAll allows LOAST to run the Positive command on each API in the config file iteratively, removing the bottleneck process and reducing the parameters required to be passed to it.

Unit tests have been added for the new PositiveAll command. Please conduct a thorough review.

### CODE REVIEW

File: test/commands/positive.test.ts [hunk 1]
```diff
@@ -421,7 +421,7 @@ describe('Positive', () => {
 
       await Positive.run(['https://url-does-not-matter.com']);
 
-      expect(mockPrompt).toHaveBeenCalledTimes(1);
+      expect(mockPrompt).toHaveBeenCalledTimes(2);
```
Comment (peyton-lucas) [inline test/commands/positive.test.ts:424]: Please verify that this is ok.

**Comment by peyton-lucas** [comment]
Good catch. Yeah no I just forgot to remove that commented code.

File: test/commands/positive.test.ts (hunk 1 above)
Comment (mheusgit) [reply test/commands/positive.test.ts:424]: The other mockPrompt further down the code has a similar 2x call.

File: src/commands/positive.ts [hunk 2]
```diff
@@ -250,20 +256,27 @@ export default class Positive extends Command {
...
+            }));
+        }
+        // if apiKey is undefined, throw an error and provide a message to the user
+        apiKeyValue = apiKey;
         if (!apiKey) {
           apiKey = apiKeyValue;
+          this.log('API key is undefined or empty.');
         }
 
         this.securityValues[apiSecurityName] = { value: apiKeyValue };
```
Comment (christineyost) [inline src/commands/positive.ts:282]:
```suggestion
        if (!noPrompt && !apiKey) {
          apiKey = // eslint-disable-next-line no-await-in-loop
            (await cli.prompt('Please provide your API Key', {
              type: 'mask',
            }));
          }
 
        // if apiKey is undefined, throw an error and provide a message to the user
        if (!apiKey) {
          this.log('API key is undefined or empty.');
        }

        this.securityValues[apiSecurityName] = { value: apiKey };
```

Comment (christineyost) [reply src/commands/positive.ts:282]: The formatting of this suggestion is bad, but I think we need something like this.

The way it's currently written if the API key flag isn't passed in, then the API key value stays undefined even if the user provides one

**Review by mheusgit — APPROVED** [review]
Looks good to me.

File: src/commands/positive.ts [hunk 3]
```diff
@@ -250,20 +256,27 @@ export default class Positive extends Command {
...
+            // eslint-disable-next-line no-await-in-loop
+            (await cli.prompt('Please provide your API Key', {
+              type: 'mask',
+            }));
+        }
+        // if apiKey is undefined, throw an error and provide a message to the user
+        apiKeyValue = apiKey;
         if (!apiKey) {
           apiKey = apiKeyValue;
+          this.log('API key is undefined or empty.');
```
Comment (christineyost) [inline src/commands/positive.ts:279]: I think we should consider throwing an error here and using oclif's error handling https://oclif.io/docs/error_handling

File: src/commands/positive.ts [hunk 4]
```diff
@@ -275,14 +288,20 @@ export default class Positive extends Command {
...
+            (await cli.prompt('Please provide your token', {
+              type: 'mask',
+            }));
+        }
+        tokenValue = token;
+
         if (!token) {
           token = tokenValue;
+          this.log('Bearer token is undefined or empty.');
         }
```
Comment (christineyost) [inline src/commands/positive.ts:305]: Same issue here as with the API key. This is why that test is suddenly calling mockPrompt twice

**Review by christineyost — CHANGES_REQUESTED** [review]
Some current functionality with security values isn't working with these changes

File: src/commands/positive.ts (hunk 3 above)
Comment (christineyost) [reply src/commands/positive.ts:279]: This should possibly be done as a separate ticket though... I think refactoring will make this easier

**Comment by peyton-lucas** [comment]
Security functionality has been restored. Thanks for the assist.

File: src/commands/positive.ts (hunk 3 above)
Comment (christineyost) [reply src/commands/positive.ts:284]: Started a ticket here: https://vajira.max.gov/browse/API-14957

Comment (christineyost) [reply src/commands/positive.ts:284]: Started a ticket here: https://vajira.max.gov/browse/API-14957

**Review by christineyost — APPROVED** [review]
Thanks for updating. Looks good to me!

**Pull request merged**